infrastructure-scripts/Create-SqlDbServiceAccount.ps1

```diff
@@ -0,0 +1,139 @@
+  $agentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+  New-AzureRmSqlServerFirewallRule -StartIPAddress $agentIp -EndIPAddress $agentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+function Update-AzureSQLServerFirewallRule{
+  $agentIP= (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+  Set-AzureRmSqlServerFirewallRule -StartIPAddress $agentIp -EndIPAddress $agentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+
+function Get-RandomPassword{
+  $Password = ([char[]]([char]33..[char]95) + ([char[]]([char]97..[char]126)) + 0..9 | Sort-Object {Get-Random})[0..33] -join ''
+  return $Password
+}
+function Invoke-SqlQuery([String] $query) {
+  $output = Invoke-Sqlcmd -ServerInstance $ServerFQDN -Database $DataBaseName -Username $SqlUserName -Password $SqlPassword -OutputSqlErrors $true -Query $query
+  return $output
+}
+
+function Remove-AzureSQLServerFirewallRule {
+   If ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue))
+    {
+      Remove-AzureRmSqlServerFirewallRule -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+    }
+}
+
+try {
+$AccountExistQuery = "SELECT * 
+FROM sys.database_principals
+WHERE name = '$SqlServiceAccountName'"
+
+
+$secretName = "${Enviroment}-${SqlServiceAccountName}"
+
+Write-host "$secretName"
+
+If ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue) -eq $null)
+{
+  New-AzureSQLServerFirewallRule
+}
+else
+{
+  Update-AzureSQLServerFirewallRule
+}
+
+Write-host "$ServerName.database.windows.net"
+$ServerFQDN = "$ServerName.database.windows.net"
+$accountExist = Invoke-SqlQuery $AccountExistQuery 
+Write-Output $accountExist
+
+
+if ($accountExist) {
+  $AccountPassword = Get-AzureKeyVaultSecret -VaultName $KeyVaultName -name $secretName 
+  $query = "ALTER USER `"$SqlServiceAccountName`" WITH PASSWORD = `'$AccountPassword`' "
+  Invoke-SqlQuery -query $query
+}
+else {
+  $AccountPassword = Get-RandomPassword 
+  $aecure = $AccountPassword  |ConvertTo-SecureString -AsPlainText -Force 
+  Set-AzureKeyVaultSecret -VaultName $KeyVaultName -Name $secretName -SecretValue $aecure 
+  $query = "CREATE USER `"$SqlServiceAccountName`"  WITH PASSWORD = `'$AccountPassword`' "
+ Invoke-SqlQuery -query $query 
+  $query2 = "ALTER ROLE db_datareader ADD MEMBER `"$SqlServiceAccountName`" "
+  Invoke-SqlQuery -query $query2
+  $query3 = "ALTER ROLE db_datawriter ADD MEMBER `"$SqlServiceAccountName`" "
+  Invoke-SqlQuery -query $query3
+  $query4 = "GRANT EXECUTE TO `"$SqlServiceAccountName`" "
+  Invoke-SqlQuery -query $query4
```

Comment:
Can we use a here-string to combine the above into a single query?

Reply:
Done